### PR TITLE
ci: run command according to krew latest releases

### PR DESCRIPTION
**Pull request by subhamkrai**
Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

File: .github/workflows/ci-for-default-ns.yaml [hunk 1]
```diff
@@ -28,10 +28,44 @@ jobs:
...
           kubectl rook-ceph ceph status --format json-pretty
-          kubectl rook-ceph ceph mon stat
+          POD=$(kubectl -n rook-ceph get pod -l app=rook-ceph-operator -o jsonpath="{.items[0].metadata.name}")
+          kubectl rook-ceph operator restart
+
+          # let's wait for operator pod to be restart
+          kubectl -n rook-ceph wait --for=delete pod/$POD --timeout=100s
+          tests/github-action-helper.sh wait_for_operator_pod_to_be_ready_state_default
+          kubectl rook-ceph operator set ROOK_LOG_LEVEL DEBUG
+          kubectl rook-ceph --context=$(kubectl config current-context) mons
```
Comment (travisn) [inline .github/workflows/ci-for-default-ns.yaml:41]: Why do we need the kubectl context here and not for other commands?

File: .github/workflows/ci-for-diff-ns.yaml [hunk 2]
```diff
@@ -25,48 +23,73 @@ jobs:
           op-ns: "test-operator"
           cluster-ns: "test-cluster"
 
-      - name: install script
-        run: sudo install kubectl-rook-ceph.sh /usr/local/bin/kubectl-rook_ceph
+      - name: install krew
+        run: tests/github-action-helper.sh install_krew
 
-      - name: Test Plugin
+      - name: Test pluging with ceph commands
```
Comment (travisn) [inline .github/workflows/ci-for-diff-ns.yaml:29]:
```suggestion
      - name: Test plugin with ceph commands
```

File: .github/workflows/ci-for-diff-ns.yaml [hunk 3]
```diff
@@ -25,48 +23,73 @@ jobs:
...
           # osd
-          kubectl rook_ceph -o test-operator -n test-cluster debug start rook-ceph-osd-0
+          kubectl rook-ceph -o test-operator -n test-cluster debug start rook-ceph-osd-0
           # sleep till the debug pod get running
           sleep 5
-          kubectl rook_ceph -o test-operator -n test-cluster debug stop rook-ceph-osd-0
+          kubectl rook-ceph -o test-operator -n test-cluster debug stop rook-ceph-osd-0
+  # This test is required to test latest changes or the changes that not present
+  # with current version of rook-ceph krew plugin
+  with-pr-changes-in-custom-namespace:
```
Comment (travisn) [inline .github/workflows/ci-for-diff-ns.yaml:72]: This is a difficult pattern to maintain if every new feature is added to this section, then it merges, then included in a release, then we need a separate PR to move it to the other section. What if we just leave everything in the CI in the non-official krew commands using `rook_ceph`, and then no need to update the CI later after new features are added? Or is there a benefit of converting the CI like this other than using the exact syntax as the docs?

**Review by travisn — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-for-diff-ns.yaml (hunk 3 above)
Comment (subhamkrai) [reply .github/workflows/ci-for-diff-ns.yaml:72]: no, I don't think we have a benefit here. So, should we close this pr or any other changes you are suggesting?

Comment (travisn) [reply .github/workflows/ci-for-diff-ns.yaml:72]: Yeah how about if we close this PR and leave the tests with their current pattern

**Pull request closed without merging**